.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: aiodynamo release workflow
+
+on:
+  release:
+    types: [released] # only triggered on release, not pre-release or draft release
+
+permissions:
+  contents: read
+  id-token: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  PYTHON_VERSION_MIN: 3.8
+  TARGET_BRANCH: ${{ github.event.release.target_commitish }}
+  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
```

Comment:
Super-minor and debatable: since github-script is used anyway, why not evaluate the event bits directly in the script block?

Reply:
The script block is there really just to early-exit the workflow, since it's the easiest way to do it at the moment
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#:~:text=echo-,core.setFailed,-Used%20as%20a

Otherwise, I think it's better to keep all the logic in GitHub Action yaml and not in github-script.